clients/retrieval_client.go

```diff
@@ -79,6 +82,8 @@ func (r *retrievalClient) RetrieveBlob(
 			r.logger.Warn("failed to dial operator while fetching BlobHeader, trying different operator", "operator", opInfo.Socket, "err", err)
 			continue
 		}
+		// Set Operator as available
+		availableOperators[opID] = true
```

Comment:
We're adding available operators here if they successfully return blob header, but marking operators as unavailable (in L170) when they fail to return chunks. Those are two different endpoints though.
Should we use the same endpoint to mark them as available and unavailable?